src/magic_enchantment.cpp

```diff
@@ -0,0 +1,361 @@
+        JsonArray jarray = jo.get_array( "effects" );
+        while( jarray.has_more() ) {
+            JsonObject effect_obj;
+            time_duration dur = read_from_json_string<time_duration>( *effect_obj.get_raw( "frequency" ),
+                                time_duration::units );
+            if( effect_obj.has_array( "spell_effects" ) ) {
+                JsonArray jarray = effect_obj.get_array( "spell_effects" );
+                while( jarray.has_more() ) {
+                    fake_spell fake;
+                    JsonObject fake_spell_obj = jarray.next_object();
+                    fake.load( fake_spell_obj );
+                    add_activation( dur, fake );
+                }
+            } else if( effect_obj.has_object( "spell_effects" ) ) {
+                fake_spell fake;
+                JsonObject fake_spell_obj = effect_obj.get_object( "spell_effects" );
+                fake.load( fake_spell_obj );
+                add_activation( dur, fake );
+            }
+        }
+    }
+
+    active_conditions.first = io::string_to_enum<has>( jo.get_string( "has", "HELD" ) );
+    active_conditions.second = io::string_to_enum<condition>( jo.get_string( "condition",
+                               "ALWAYS" ) );
+
+    if( jo.has_array( "values" ) ) {
+        JsonArray jarray = jo.get_array( "values" );
+        while( jarray.has_more() ) {
+            JsonObject value_obj = jarray.next_object();
+            const enchantment::mod value = io::string_to_enum<mod>( value_obj.get_string( "value" ) );
+            const int add = value_obj.get_int( "add", 0 );
+            const double mult = value_obj.get_float( "multiply", 0.0 );
+            if( add != 0 ) {
+                values_add.emplace( value, add );
+            }
+            if( mult != 0.0 ) {
+                values_multiply.emplace( value, mult );
+            }
+        }
+    }
+}
+
+void enchantment::serialize( JsonOut &jsout ) const
+{
+    jsout.start_object();
+
+    jsout.member( "has", io::enum_to_string<has>( active_conditions.first ) );
+    jsout.member( "condition", io::enum_to_string<condition>( active_conditions.second ) );
+
+    if( !hit_you_effect.empty() ) {
+        jsout.member( "hit_you_effect" );
+        jsout.start_array();
+        for( const fake_spell &sp : hit_you_effect ) {
+            sp.serialize( jsout );
+        }
+        jsout.end_array();
+    }
+
+    if( !hit_me_effect.empty() ) {
+        jsout.member( "hit_me_effect" );
+        jsout.start_array();
+        for( const fake_spell &sp : hit_me_effect ) {
+            sp.serialize( jsout );
+        }
+        jsout.end_array();
+    }
+
+    if( !intermittent_activation.empty() ) {
+        jsout.member( "intermittent_activation" );
+        jsout.start_object();
+        for( const std::pair<time_duration, std::vector<fake_spell>> pair : intermittent_activation ) {
+            jsout.member( "duration", pair.first );
+            jsout.start_array( "effects" );
```

Comment:
That is not a correct way to name an array member - `start_array` does not take string parameter for name, only Boolean for wrapping.

![image](https://user-images.githubusercontent.com/16213433/91308878-93545e00-e7b8-11ea-8d40-62c4d9368f1f.png)

You probably need to do something like this to make it work properly:

```CPP
            jsout.member( "effects" );
            jsout.start_array();
```